### PR TITLE
Updates for boolean fields

### PR DESCRIPTION
**Pull request by jcal-15**
- adding support for default values of True
- adding support to require a user to specify the value

Also, Adding Python 3.10 to the list of support Python versions and adding testing.

### CODE REVIEW

File: tests/test_functional.py [hunk 1]
```diff
@@ -31,7 +30,7 @@ def __exit__(self, exc_type, exc_value, traceback):
         return exc_type is RuntimeError and str(exc_value) == "App Exited"
 
 
-class FunctionalParserTests(unittest.TestCase):
+class FunctionalParserTests(unittest.TestCase):
```
Comment (mivade) [inline tests/test_functional.py:33]: Nitpick: can you please remove the trailing whitespace?

**Review by mivade — COMMENTED** [review]
Looks great, thanks for doing this! My extremely minor nitpick is the one line with trailing whitespace. I haven't been giving this project much attention lately but I should really add some linting to the tests.

File: tests/test_functional.py (hunk 1 above)
Comment (jcal-15) [reply tests/test_functional.py]: No problem. I ran black on the test files to make sure they conformed to the standards. 
Would you like me to add that the to the tox configuration too?

Comment (mivade) [reply tests/test_functional.py]: Thanks! If you don't mind adding a `black --check` call to the tox configuration that would be appreciated.

Comment (jcal-15) [reply tests/test_functional.py]: odd it is failing on GitHub, but it passed locally.

File: tox.ini [hunk 2]
```diff
@@ -10,3 +10,4 @@ commands =
     pytest
 commands_post =
     black --check argparse_dataclass.py
+    black --check {toxinidir}/tests/*.py
```
Comment (mivade) [inline tox.ini]: Maybe try combining this line with the above:

```
black --check argparse_dataclass.py tests/
```

I'm not really sure why this was erroring as is though.

File: tests/test_functional.py (hunk 1 above)
Comment (jcal-15) [reply tests/test_functional.py]: Got it to work..
Odd issue it must be that that wildcards work on Windows but not Linux. 
Checking the directory is a cleaner approach anyways.

File: tox.ini (hunk 2 above)
Comment (jcal-15) [reply tox.ini]: Yes, that turned out to be the issue. Thanks.

**Comment by mivade** [comment]
Thanks!

**Pull request merged**